spec/solver_spec.rb

```diff
@@ -0,0 +1,76 @@
+require_relative 'spec_helper'
+
+RSpec.describe Solver do
+  # context 'checks to make sure that Solver is initialized' do
+
+  # end
+  context 'Testing the factorial method' do
+    it 'Checks that the factorial method can receive a value' do
+      mock = double('solver')
+      allow(mock).to receive(:factorial).with(0)
+
+      solver = Solver.new
+      solver.factorial(1)
+    end
+
+    it 'Checks that the factorial method only works with positive numbers including 0' do
+      solver = Solver.new
+      expect(solver.factorial(-1)).to eql 'invalid number'
+    end
+
+    it 'Checks that the factorial method identifies 0 as an argument and returns 1' do
+      solver = Solver.new
+      expect(solver.factorial(0)).to eql 1
+    end
+
+    it 'Checks that the factorial method returns the correct factorial' do
+      solver = Solver.new
+      expect(solver.factorial(2)).to eql 3
+    end
```

Comment:
- [ ] The method of `factorial` is not looking right. As in this we multiply from 1 upto the number which we have to find the factorial.

`Hint: factorial(5) - 1 * 2 * 3 * 4 * 5 = 120

```
factorial(2) - we would expect 2.
factorial(5) - we would expect 120.
```

Please fix this test and method.